core/state/snapshot/iterator_test.go

```diff
@@ -134,10 +194,12 @@ func verifyIterator(t *testing.T, expCount int, it AccountIterator) {
 		if bytes.Compare(last[:], hash[:]) >= 0 {
 			t.Errorf("wrong order: %x >= %x", last, hash)
 		}
-		if it.Account() == nil {
+		count++
+		if verify == verifyAccount && it.(AccountIterator).Account() == nil {
+			t.Errorf("iterator returned nil-value for hash %x", hash)
+		} else if verify == verifyStorage && it.(StorageIterator).Slot() == nil {
```

Comment:
Might be good to check not only against `nil`, but also count empty slice as nil?